src/ir/graph.partition.cpp

```diff
@@ -49,25 +50,30 @@ struct region
             for (auto out : n.outputs())
                 outputs.emplace(out);
 
-            std::unordered_map<output_connector *, int> need_remove_outputs;
             for (auto it = region_inputs.begin(); it != region_inputs.end();)
             {
                 if (outputs.contains((*it)->connection()))
                 {
                     if (need_remove_outputs.find((*it)->connection()) != need_remove_outputs.end())
-                        need_remove_outputs.at((*it)->connection()) += 1;
+                        need_remove_outputs.at((*it)->connection()) -= 1;
                     else
-                        need_remove_outputs.emplace((*it)->connection(), 1);
+                        need_remove_outputs.emplace((*it)->connection(),
+                            (*it)->connection()->connections().size() - 1);
                     it = region_inputs.erase(it);
                 }
                 else
                     ++it;
             }
 
-            for (auto it : need_remove_outputs)
+            for (auto it = need_remove_outputs.begin(); it != need_remove_outputs.end();)
             {
```

Comment:
If the output has not only one connection, it should be deleted until all the connections of the "output" join this region. So here should have a countor.